### ISSUE TITLE
🐛 Bug Report: <After upgrading to 1.31, Login failed; caused by Error: Login failed, user profile does not contain an email>

### ISSUE THREAD

**Issue by ankitapandey1106** [backstage/backstage#27971]
### 📜 Description

Trying to login to EIDP, using oauth provider and a custom sign in resolver. It fails with this error - Login failed; caused by Error: Login failed, user profile does not contain an email
It was working fine till 1.20 upgrade but we are upgrading now to 1.31 with new backend system and new auth and it fails to login

### 👍 Expected behavior

It should let the user login

### 👎 Actual Behavior with Screenshots

getting this error - Login failed; caused by Error: Login failed, user profile does not contain an email
We are using the approach where user is not in backstage.
![image](https://github.com/user-attachments/assets/82e95f80-3fa6-4bd2-9f47-b280f032bfa3)

### 👟 Reproduction steps

When the user tries to login

### 📃 Provide the context for the Bug.

Pasting below the custom code we have done for sign in resolver in our index.ts of backend package.
![image](https://github.com/user-attachments/assets/ad91231d-ca03-4725-9e60-6528e9f3d3b2)

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

**Comment by ankitapandey1106** [comment]
HI @Rugvip / @freben  Any help would be appreciated, we are stuck with our upgrade cuz of this issue

**Comment by camilaibs** [comment]
Hi @ankitapandey1106 could you please run `yarn tsc` and share if this command logs any error?

**Comment by ankitapandey1106** [comment]
> Hi @ankitapandey1106 could you please run `yarn tsc` and share if this command logs any error?

Hi @camilaibs there are no errors in tsc, it's absolutely fine. What I see in logs is the profile is null in below code of index file of backend package. do you think we need to set it explicitly somewhere ? earlier we were setting it in legacy code in auth. I will paste the snippet below 
factory: createOAuthProviderFactory({
            authenticator: oauth2Authenticator,
            async signInResolver(info, ctx) {
              console.log("profile ---------- " + JSON.stringify(info.profile));
              console.log("ctx ---------- " + JSON.stringify(ctx));
              if (!info.profile.email) {
                throw new Error(
                  'Login failed, user profile does not contain an email',
                );
              }
              // Split the email into the local part and the domain.
              const [localPart] = info.profile.email.split('@');

we are setting like this in all the legacy classes of lib inside auth plugin- 
![image](https://github.com/user-attachments/assets/e5356fb6-89d7-408b-bbc3-fb68b5c4547b)

**Comment by camilaibs** [comment]
Could you also share what the `result` object contains?
![image](https://github.com/user-attachments/assets/e270b7b0-3631-4a62-a08b-a7d59bb381a6)

**Comment by ankitapandey1106** [comment]
Hi @camilaibs  it's coming as null, that's what I saw in logs

**Comment by Rugvip** [comment]
@ankitapandey1106 it's possible that your profile doesn't contain any information if there's no immediately discoverable information in the response from Okta. It looks like you where parsing the access token before to get the email, and it might be that you need to do that still. It'll be something like `decodingJWT(info.result.session.accessToken).email` where `info` is the first parameter of the `signInResolver` callback.

**Comment by ankitapandey1106** [comment]
Hi @Rugvip  that's right, we are parsing the access token to get the email, using something like this - decodingJWT(info.result.session.accessToken).email in legacysigninresolver, and that code still exists.
do we need to parse it in the index file as well after the new backend and new auth system? like in here where we are reading the info in signinresolver- ?
const customAuth = createBackendModule({
  // This ID must be exactly "auth" because that's the plugin it targets
  pluginId: 'auth',
  // This ID must be unique, but can be anything
  moduleId: 'oauth2',
  register(reg) {
    reg.registerInit({
      deps: { providers: authProvidersExtensionPoint },
      async init({ providers }) {
        providers.registerProvider({
          // This ID must match the actual provider config, e.g. addressing
          // auth.providers.github means that this must be "github".
          providerId: 'oauth2',
          // Use createProxyAuthProviderFactory instead if it's one of the proxy
          // based providers rather than an OAuth based one
          factory: createOAuthProviderFactory({
            authenticator: oauth2Authenticator,
            async signInResolver(info, ctx) {
              console.log("profile ---------- " + JSON.stringify(info.profile));
              console.log("ctx ---------- " + JSON.stringify(ctx));
              if (!info.profile.email) {
                throw new Error(
                  'Login failed, user profile does not contain an email',
                );
              }
              // Split the email into the local part and the domain.
              const [localPart] = info.profile.email.split('@');

**Comment by Rugvip** [comment]
@ankitapandey1106 yes the new module replaces the legacy provider, they don't communicate with each other in any way.

**Comment by ankitapandey1106** [comment]
Suggested fix worked
